### DISCUSSION TITLE
About the different patterns for citekeys described in the document.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [retorquere/zotero-better-bibtex#2502] [General]
I noticed the following description [here](https://retorque.re/zotero-better-bibtex/installation/):

> Note: the default setting of BBT will generate different citekeys than Zotero would itself generate; the keys from Zotero are not always safe for use in bibtex/biber. If you want to get the stock zotero keys, set the pattern in the preferences to [zotero]. I very much recommend not choosing [zotero] as your pattern unless you have existing articles that use keys generated by previous Zotero-native BibTeX export.

I'm not sure if the above description refers to the corresponding setting on the panel below:

![image](https://user-images.githubusercontent.com/11155854/235331883-f3157e77-2222-4224-b7c0-de4faa9a87cb.png)

Any clues for my ambiguous understanding will be appreciated.

Regards,
Zhao

**Comment by retorquere** [comment]
It does refer to that setting.

**Reply by hongyi-zhao** [reply]
Why must it be written as `[zotero]` instead of `zotero`?

**Reply by retorquere** [reply]
Oh I need to update the docs. `zotero` is the proper format now, but BBT will change `[zotero]` to `zotero` automatically.

**Reply by hongyi-zhao** [reply]
Why are there redundant `[]`?

**Reply by retorquere** [reply]
Because they used to be required. I will update the docs.

**Reply by hongyi-zhao** [reply]
Do you mean, they are not required now, aka, `[bla]` is not used now, instead `bla` should be used?

**Reply by retorquere** [reply]
Mostly yes. If you type the old format it should be updated automatically, but I see now that gets stuck in the conversion. I'll work on a fix for that, but by and large, you can type `auth.lower + year` where you used to type `[auth:lower][year]`. There are a few situations where that will not work, but the error message under the edit field will tell you what's wrong.

**Reply by hongyi-zhao** [reply]
IMO, all the citekeys are generated automatically when the ref is imported. Basically, users don't need and shouldn't care about the underlying mechanism.

**Reply by retorquere** [reply]
Then leave it at the default and ignore it? I don't understand why you have an opinion on whether other people should care about these things. I know for a fact that people *do* care about this as I get plenty of questions about it on how to use it.

**Reply by hongyi-zhao** [reply]
> Then leave it at the default and ignore it?

Keep it as simple as possible!
 
> I don't understand why you have an opinion on whether other people should care about these things. 

I was speaking from the perspective of a general user, who may not necessarily have in-depth knowledge of computer programming. Therefore, they may not necessarily pay attention to this matter. Even if they notice it, they may not understand what it means, just like myself.

> I know for a fact that people do care about this as I get plenty of questions about it on how to use it.

I also care about it because I want to have a unified solution and clear up possible confusion to avoid potential misuse. However, I believe that only a small fraction of the total number of users who use this tool are the ones who would be discussing this issue with you.

**Reply by retorquere** [reply]
> > Then leave it at the default and ignore it?
> 
> Keep it as simple as possible!

As simple as possible is to use the default bibtex functionality included in Zotero. Or you could leave all BBT settings at default and ignore them, which seems simple enough to me. BBT will work just fine that way.
 
> > I don't understand why you have an opinion on whether other people should care about these things.
> 
> I was speaking from the perspective of a general user

I think I'm in a decent position to know what the general BBT user wants by now, and being able to generate citekeys according to a formula is what many people want.

I understand wanting to have a good out-of-the-box experience. But it seems pretty wild to me to claim on behalf of users you don't know that they *should not* care about these things. They *should not have to care* about these things, and they don't have to. But if they do care, why should they not?

> who may not necessarily have in-depth knowledge of computer programming. Therefore, they may not necessarily pay attention to this matter. 

**Then don't**. This is fine. BBT at its default settings works just fine.

> Even if they notice it, they may not understand what it means, just like myself.

Asking is always fine.

> I also care about it because I want to have a unified solution and clear up possible confusion to avoid potential misuse. However, I believe that only a small fraction of the total number of users who use this tool are the ones who would be discussing this issue with you.

And it's fine if they just leave everything about BBT at their defaults. You don't *have* to go into the preferences.

**Reply by retorquere** [reply]
A new version is out that will show and use the new format when an old format is entered. You can safely ignore this if you want. It's there for others.

**Reply by hongyi-zhao** [reply]
> Or you could leave all BBT settings at default and ignore them, which seems simple enough to me.

IIRC, you've said that this is a more robust method than the default bibtex functionality included in Zotero. So, I prefer this one.

> But if they do care, why should they not?

Yes, they can certainly be interested and further clarify any confusion through questioning and discussion, reaching consensus, which can further promote and accelerate the development process of the tool, which is a very good thing.

N.B.: Ever since we had a conversation a long time ago, I have been deeply impressed by your professionalism as a developer of this package, and I greatly admire it.
